collector/internal/controller.go

```diff
@@ -24,15 +24,33 @@ const (
 // Controller is a bridge between the Collector's [receiverprofiles.Profiles]
 // interface and our [internal.Controller]
 type Controller struct {
-	ctlr *controller.Controller
+	ctlr       *controller.Controller
+	onShutdown func()
+}
+
+// Option is a function that allows to configure a ControllerOption.
+type Option func(*ControllerOption)
+
+// ControllerOption is the extra configuration for the controller.
+type ControllerOption struct {
```

Comment:
I'm not sure this needs to be a public struct (it's internal, granted but still).